File: Leetcode/Binary Tree Inorder Traversal/Binary Tree Inorder Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root):

        if not root:
            return root
        
        res = []                                            # Declaring the list to contain tree elements            
            
        if root.left:
            res += self.inorderTraversal(root.left)         # Updating list to find if the tree is left
        
        if root:                                            # Appending value of root in the list
            res.append(root.val)
        
        if root.right:
            res += self.inorderTraversal(root.right)        # Updating list to find if the tree is right

        return res                                          # Returning result
        

File: Leetcode/Binary Tree Inorder Traversal/Binary Tree Inorder Traversal.java
// Given the root of a binary tree, return the inorder traversal of its nodes' values.
import java.util.*;
class Solution {
    // Initialize an arraylist
    List<Integer> al = new ArrayList();
    public List<Integer> inorderTraversal(TreeNode root) {
        iot(root);
       return al; 
    }
    // iot is a helper method to store the inorder traversed node value in ArrayList al.
    public void iot(TreeNode root){
        // Base condition: root != null
        if(root != null){
            // Traverse the left subtree of root
            if(root.left != null){
                iot(root.left);
            }
            // root add root value to the list
            al.add(root.val);
            // Traverse the right subtree of root
            if(root.right != null){
                iot(root.right);
            }
        }
        
    }
}